core/arch/arm/kernel/tee_time.c

```diff
@@ -57,6 +57,17 @@ void tee_time_wait(uint32_t milliseconds_delay)
 	thread_rpc_cmd(OPTEE_MSG_RPC_CMD_SUSPEND, 1, &params);
 }
 
+void tee_time_mdelay(uint32_t ms)
+{
+	TEE_Time prev, curr;
+
+	tee_time_get_sys_time(&prev);
+
+	do {
+		tee_time_get_sys_time(&curr);
+	} while (curr.millis - prev.millis < ms);
```

Comment:
Yes, was researching the overflow case when you replied. The macros were indeed helpful. Thanks!